@here/olp-sdk-core/lib/utils/BlobData.ts

```diff
@@ -38,7 +38,7 @@ export abstract class BlobData {
     /**
      * Gets the size of the data.
      *
-     * @returns The length of the data in bytes.
+     * @returns The `Promise` object with the length of the data in bytes.
      */
-    abstract size(): number;
+    abstract size(): Promise<number>;
```

Comment:
I dont think that a promise for such a simple API is necessary. I would just return 0 in case the file path is invalid or the file could not be opened.

Reply:
We don't have possibility to use async operations in constructor. In case of we calling the size method first time (we don't have cached file descriptor), we should open the file in async way

Reply:
you are right. I changed the approach. I've added a build method and refactored that. It's because we can't use async operations in the constructor